pandas/_libs/tslibs/src/datetime/np_datetime_strings.c

```diff
@@ -66,10 +66,24 @@ This file implements string parsing and creation for NumPy datetime.
  *
  * Returns 0 on success, -1 on failure.
  */
+
+#define FORMAT_STARTSWITH(ch) \
```

Comment:
Can you make this a function instead of a macro?

Reply:
The macro modifies two variables and has access to the goto location. A separate function then has to have at least two variables accepted with one of them being a pointer to a pointer. It also has to return a flag that needs a check every single time it is called. We can remove this pointer to a pointer, however, and modify just an offset variable, but checks are unavoidable. Is that a preferred approach? What do you think about checks being done in a macro instead?

Reply:
You can also just do one function that performs the check and in the caller run the goto in the failing branch while advancing the pointers in the succeeding branch. It is more verbose but also more explicit than the macro.